### PR TITLE
Cleanup and documenation

### PR DESCRIPTION
**Pull request by jasagredo**
- Renamed `prologue.md` to `prologue.haddock`. It is a haddock file not a markdown one. Rephrase some parts.

- Regenerate the module map and provide a command to do so again.

- Fixed broken links (used https://github.com/serokell/xrefcheck).

- Re-format some lines to 80 chars.

- Make `haddocks.sh` report no ShellCheck errors.

- Fix haddock contents redirection.

- Nicer output in `haddocks.sh`.

- Include test-suites, benchmarks, executables in haddock output.

### CODE REVIEW

**Comment by jasagredo** [comment]
The nicer output 
image

File: scripts/docs/haddocks.sh [hunk 1]
```diff
@@ -1,38 +1,35 @@
 #!/bin/bash
```
Comment (amesgen) [inline scripts/docs/haddocks.sh:1]: As `/bin/bash` does not exist on NixOS:
```suggestion
#!/usr/bin/env bash
```
so I no longer have to run `bash scripts/docs/haddocks.sh` :laughing:

File: README.md [hunk 2]
```diff
@@ -7,10 +7,9 @@
 [![sop-extras](https://img.shields.io/badge/sop--extras-0.1.0.0-blue)](https://input-output-hk.github.io/cardano-haskell-packages/package/sop-extras-0.1.0.0/)
 [![strict-sop-core](https://img.shields.io/badge/strict--sop--core-0.1.0.0-blue)](https://input-output-hk.github.io/cardano-haskell-packages/package/strict-sop-core-0.1.0.0/)
 
-[![docs](https://img.shields.io/badge/Documentation-yellow)](https://input-output-hk.github.io/ouroboros-consensus/)
+[![docs](https://img.shields.io/badge/Documentation-yellow)][webpage]
 
-Implementation of the [Ouroboros-family](docs/website/README.md) of consensus
```
Comment (amesgen) [inline README.md:12]: I considered at least twice per week opening a PR just to remove this hyphen :pray:

**Review by amesgen — APPROVED** [review]
Thanks, also ran `./scripts/docs/haddocks.sh` locally and confirmed that it works well :+1:

File: scripts/docs/haddocks.sh (hunk 1 above)
Comment (amesgen) [reply scripts/docs/haddocks.sh:1]: Oh, the PR is already in the merge queue; this change isn't too important, so I won't stop it.

**Pull request merged**